test/framework/modules.py

```diff
@@ -489,15 +489,23 @@ def test_curr_module_paths(self):
         test3 = os.path.join(self.test_prefix, 'test3')
         mkdir(test3)
 
+        del os.environ['MODULEPATH']
+        self.assertEqual(curr_module_paths(), [])
+        self.assertEqual(curr_module_paths(clean=False), [])
+
         os.environ['MODULEPATH'] = ''
         self.assertEqual(curr_module_paths(), [])
+        self.assertEqual(curr_module_paths(clean=False), [''])
 
-        os.environ['MODULEPATH'] = '%s:%s:%s' % (test1, test2, test3)
+        os.environ['MODULEPATH'] = os.pathsep.join([test1, test2, test3])
         self.assertEqual(curr_module_paths(), [test1, test2, test3])
+        self.assertEqual(curr_module_paths(clean=False), [test1, test2, test3])
 
         # empty entries and non-existing directories are filtered out
-        os.environ['MODULEPATH'] = '/doesnotexist:%s::%s:' % (test2, test1)
+        os.environ['MODULEPATH'] = os.pathsep.join(['/doesnotexist', test2, '', test1, ''])
```

Comment:
strictly speaking it should be
os.sep + 'doesnotexist'
but we're failing on that account in soooo many places...